drivers/wireless/lpwan/rn2483/rn2483.c

```diff
@@ -334,56 +346,210 @@ static int rn2483_close(FAR struct file *filep)
 static ssize_t rn2483_read(FAR struct file *filep, FAR char *buffer,
                            size_t buflen)
 {
-  syslog(LOG_INFO, "Enter read\n");
-
   FAR struct inode *inode = filep->f_inode;
   FAR struct rn2483_dev_s *priv = inode->i_private;
   ssize_t length = 0;
+  ssize_t received;
   int err;
+  char response[30] = {0}; // TODO: shortest length?
+
+  /* If file position is non-zero but we're marked as no longer receiving,
+   * then the whole transmission has been consumed. Set a filepos of 0 to
+   * signal EOF.
+   */
+
+  if (priv->receiving == false && filep->f_pos > 0)
+    {
+      filep->f_pos = 0;
+      return 0;
+    }
 
-  /* If file position is non-zero, then we're at the end of file. */
+  /* We're setting a new receive call, mark as receiving */
 
-  if (filep->f_pos > 0)
-  {
-    return 0;
-  }
+  priv->receiving = true;
 
   /* Get exclusive access */
 
   err = nxmutex_lock(&priv->devlock);
   if (err < 0)
-  {
-    return err;
-  }
+    {
+      priv->receiving = false;
+      return err;
+    }
 
 #ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
   if (priv->unlinked)
-  {
-    /* Do not allow operations on unlinked drivers. */
+    {
+      /* Do not allow operations on unlinked drivers. */
 
-    nxmutex_unlock(&priv->devlock);
-    return 0;
-  }
+      priv->receiving = false;
+      nxmutex_unlock(&priv->devlock);
+      return 0;
+    }
 #endif
 
-  // TODO actually receive
+  /* Must send 'mac pause' command before every receive command */
+
+  char pause[] = "mac pause\r\n";
+  length = file_write(&priv->uart, pause, sizeof(pause));
+  if (length < 0)
+    {
+      priv->receiving = false;
+      nxmutex_unlock(&priv->devlock);
+      return length;
+    }
+
+  /* Check how the radio responded to MAC pause */
+
+  length = rn2483_read_response(priv, response, sizeof(response) - 1);
+  if (length < 0)
+    {
+      priv->receiving = false;
+      nxmutex_unlock(&priv->devlock);
+      return length;
+    }
+
+  /* We should be pausing for the max duration */
+
+  if (!strstr(response, "4294967245"))
```

Comment:
What is the significant of the number 4294967245?

Reply:
This is the amount of seconds that the MAC layer is paused for. In the datasheet it actually says it should return `4294967295`, but I checked our old code and it seems we've always had it as `4294967245`. It's also what the radio responds with so 🤷

Reply:
This could probably be a `#define` though to be less of a magic number and give it a name.